mirrord/agent/src/file.rs

```diff
@@ -483,35 +487,64 @@ impl FileManager {
 
         let path = resolve_path(path, &self.root_path)?;
 
-        match std::fs::create_dir(Path::new(&path)) {
-            Ok(_) => Ok(MakeDirResponse {
+        let c_path = match std::ffi::CString::new(path.to_string_lossy().as_bytes()) {
```

Comment:
We'd prefer to avoid libc - is it to obtain underlying error? We'd prefer to convert the errors to our "standard" (which would be Rust's probably). It's less consistent but easier to maintain in case of introducing new platforms from each side.

Reply:
This is one of the questions that I was about to ask later. I did it when I saw that the mkdirat had a custom logic for relative paths, so I just created a 1-1 match for the libc functions. I see that the same happens with `openat`  (which is already implemented), so I will use that as an example to make this change.

Regarding file/directory permissions, is there something that we should take into account?

Reply:
Nothing that comes to mind.